src/PowerShellEditorServices.Host/LanguageServer.cs

```diff
@@ -3,98 +3,76 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
-using DebugAdapterMessages = Microsoft.PowerShell.EditorServices.Protocol.DebugAdapter;
 using Microsoft.PowerShell.EditorServices.Protocol.LanguageServer;
 using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol;
+using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.Channel;
 using Microsoft.PowerShell.EditorServices.Utility;
 using Nito.AsyncEx;
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Management.Automation;
 using System.Management.Automation.Language;
 using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
-using System.IO;
+using DebugAdapterMessages = Microsoft.PowerShell.EditorServices.Protocol.DebugAdapter;
 
 namespace Microsoft.PowerShell.EditorServices.Host
 {
-    internal class LanguageServer : IMessageProcessor
+    internal class LanguageServer : LanguageServerBase 
```

Comment:
My idea was to use something like Owin and then the actual HTTP hosting could be done in the host process, Nancy or IIS. So I would think having those classes exposed would be helpful.